node/src/effect.rs

```diff
@@ -176,17 +176,20 @@ impl<T: 'static + Send> Responder<T> {
 
 impl<T> Responder<T> {
     /// Send `data` to the origin of the request.
-    pub(crate) async fn respond(mut self, data: T) {
-        if let Some(sender) = self.0.take() {
-            if sender.send(data).is_err() {
-                let backtrace = backtrace::Backtrace::new();
-                error!(
-                    ?backtrace,
-                    "could not send response to request down oneshot channel"
-                );
+    pub(crate) fn respond(mut self, data: T) -> impl Future<Output = ()> {
+        let caller_backtrace = backtrace::Backtrace::new();
+        async move {
```

Comment:
We can just un `async` this

Reply:
Turns out this is much harder than it would have appeared.

Reply:
> Turns out this is much harder than it would have appeared.

I just want to add that I totally support this decision.